refactor/src/Cage.py

```diff
@@ -815,7 +818,7 @@ def update_deltas(
             new_males: np.int64,
             new_infections: int,
             lice_from_reservoir: dict,
```

Comment:
Reminder to myself these dicts should be better typed. Also, `dict` is not a proper way to annotate!